### PR TITLE
Update dependencies and lint /2 

### PR DESCRIPTION
**Pull request by fregante**
I was able to solve the issues I had in #2391, so this updates the remaining linting dependencies to their latest version.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -117,6 +117,7 @@
 				"rules": {
 					"@typescript-eslint/no-misused-promises": "off",
 					"@typescript-eslint/promise-function-async": "off",
+					"@typescript-eslint/no-explicit-any": "error",
```
Comment (fregante) [inline package.json]: - It was disabled upstream https://github.com/xojs/eslint-config-xo-typescript/commit/bbbf88972520e8af3439ba2ab3f1a1c40e68041c
- But here its support was fixed: https://github.com/sindresorhus/refined-github/pull/2429

File: package.json [hunk 2]
```diff
@@ -107,8 +107,8 @@
 		"rules": {
 			"no-alert": "off",
 			"import/no-unassigned-import": "off",
-			"import/no-unresolved": "warn",
-			"import/named": "warn"
+			"import/no-unresolved": "off",
+			"import/named": "off"
```
Comment (fregante) [inline package.json]: Reportedly disabled upstream: https://github.com/xojs/xo/commit/ee145cbe719073aeccd349af4bb85ddc542a2818

...but it appears to still be enabled here, so I'm disabling it manually.

File: tsconfig.json [hunk 3]
```diff
@@ -6,6 +6,8 @@
 		"esModuleInterop": true
 	},
 	"include": [
-		"source"
+		"source",
+		"test",
+		"webpack.config.ts"
```
Comment (fregante) [inline tsconfig.json]: This fixed the issue I was having in the previous PR.

File: package.json (hunk 2 above)
Comment (sindresorhus) [reply package.json]: Not released yet.

**Pull request merged**